crates/matrix-sdk-base/src/read_receipts.rs

```diff
@@ -19,13 +20,12 @@ use crate::{error::Result, store::StateChanges, RoomInfo};
 pub trait PreviousEventsProvider: Send + Sync {
     /// Returns the list of known timeline events, in sync order, for the given
     /// room.
-    // TODO: return a reference or some kind of iterator
-    fn for_room(&self, room_id: &RoomId) -> Vec<SyncTimelineEvent>;
+    fn for_room(&self, room_id: &RoomId) -> Vector<SyncTimelineEvent>;
```

Comment:
Ideally, we should just return an iterator, which would allow more freedom here.